gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,192 @@
+ * that takes solution registry entries, reads their current values on the device, and returns a promise resolved to
+ * them.
+ * @param {Object} solutions - Solutions registry entries for solutions available on the current machine.
+ * @param {Object} options - Extra options for processing.
+ * @param {Array} options.solutionsList - If provided, only solutions in this list of `solutionsID`s will
+ * be captured. Example:
+ *
+ * '''json
+ * ["com.microsoft.windows.cursors", "com.freedomscientific.jaws"]
+ * '''
+ * @return {fluid.promise} Returns a promise resolving with the entire system settings capture.
+ */
+gpii.flowManager.capture.captureSystemSettings = function (readSettingsFunc, solutions, options) {
+    var solutionsToFetch = fluid.copy(solutions);
+    if (options.solutionsList) {
+        fluid.remove_if(solutionsToFetch, function (solution, solutionID) {
+            return !options.solutionsList.includes(solutionID);
+        });
+    }
+
+    return readSettingsFunc(solutionsToFetch);
+};
+
+/**
+ * The raw return payload from the capture promise sequence looks like:
+ * '''json
+ *       [
+ *           {
+ *               "fakemag1": [
+ *                   {
+ *                       "settings": {
+ *                           "magnification": 2
+ *                       }
+ *                   }
+ *               ]
+ *           },
+ *           {
+ *               "fakemag1": [
+ *                   {
+ *                       "settings": {
+ *                           "invert": true
+ *                       }
+ *                   }
+ *               ]
+ *           },
+ *           {
+ *               "fakemag2": [
+ *                   {
+ *                       "settings": {
+ *                           "magnification": 2,
+ *                           "invert": true
+ *                       }
+ *                   }
+ *               ]
+ *           }
+ *       ]
+ * '''
+ *
+ * and we want:
+ * '''json
+ *       {
+ *           "fakemag1": {
+ *               "magnification": 2,
+ *               "invert": true
+ *           },
+ *           "fakemag2": {
+ *               "magnification": 2,
+ *               "invert": true
+ *           }
+ *       }
+ * '''
+ *
+ * @param {Object} data - The raw captured data.
+ * @return {Object} Returns a new payload with collapsed data, and multiple settings handler
+ * results for the same solution merged together.
+ */
+gpii.flowManager.capture.formatRawCapturedSettings = function (data) {
+    var togo = {};
+    fluid.each(data, function (sequenceItem) {
+        if (sequenceItem.isError) {
+            fluid.log("Error capturing settings for: ", sequenceItem);
+            return;
+        }
+        fluid.each(sequenceItem, function (item, key) {
+            if (!togo[key]) {
+                togo[key] = {};
+            }
+            fluid.each(fluid.get(item, "0.settings"), function (value, settingId) {
```

Comment:
I believe @amb26 would suggest you to use more robust form of `[0, "settings"]` rather than `"0.settings"`. ;)

Reply:
Robustified :p